netbox/vpn/filtersets.py

```diff
@@ -0,0 +1,240 @@
+import django_filters
+from django.db.models import Q
+from django.utils.translation import gettext as _
+
+from dcim.models import Interface
+from ipam.models import IPAddress
+from netbox.filtersets import NetBoxModelFilterSet
+from tenancy.filtersets import TenancyFilterSet
+from utilities.filters import ContentTypeFilter, MultiValueCharFilter, MultiValueNumberFilter
+from virtualization.models import VMInterface
+from .choices import *
+from .models import *
+
+__all__ = (
+    'IKEPolicyFilterSet',
+    'IKEProposalFilterSet',
+    'IPSecPolicyFilterSet',
+    'IPSecProfileFilterSet',
+    'IPSecProposalFilterSet',
+    'TunnelFilterSet',
+    'TunnelTerminationFilterSet',
+)
+
+
+class TunnelFilterSet(NetBoxModelFilterSet, TenancyFilterSet):
+    status = django_filters.MultipleChoiceFilter(
+        choices=TunnelStatusChoices
+    )
+    encapsulation = django_filters.MultipleChoiceFilter(
+        choices=TunnelEncapsulationChoices
+    )
+    ipsec_profile_id = django_filters.ModelMultipleChoiceFilter(
+        queryset=IPSecProfile.objects.all(),
+        label=_('IPSec profile (ID)'),
+    )
+    ipsec_profile = django_filters.ModelMultipleChoiceFilter(
+        field_name='ipsec_profile__name',
+        queryset=IPSecProfile.objects.all(),
+        to_field_name='name',
+        label=_('IPSec profile (name)'),
+    )
+
+    class Meta:
+        model = Tunnel
+        fields = ['id', 'name', 'tunnel_id']
+
+    def search(self, queryset, name, value):
+        if not value.strip():
+            return queryset
+        return queryset.filter(
+            Q(name__icontains=value) |
+            Q(description__icontains=value) |
+            Q(comments__icontains=value)
+        )
+
+
+class TunnelTerminationFilterSet(NetBoxModelFilterSet):
+    tunnel_id = django_filters.ModelMultipleChoiceFilter(
+        field_name='tunnel',
+        queryset=Tunnel.objects.all(),
+        label=_('Tunnel (ID)'),
+    )
+    tunnel = django_filters.ModelMultipleChoiceFilter(
+        field_name='tunnel__name',
+        queryset=Tunnel.objects.all(),
+        to_field_name='name',
+        label=_('Tunnel (name)'),
+    )
+    role = django_filters.MultipleChoiceFilter(
+        choices=TunnelTerminationRoleChoices
+    )
+    # interface = django_filters.ModelMultipleChoiceFilter(
```

Comment:
I still need to decide on how best to work around a conflict between the `interface` GFK  and using `interface` as the reverse relationship for `dcim.Interface` to TunnelTermination. Other models (i.e. IPAddress) don't run into this because they use a different name for the local field (i.e. `assigned_object`).

Reply:
I ended up renaming `interface` to `termination`, in line with CableTermination.